packages/connectkit/src/components/DaimoPayButton/index.tsx

```diff
@@ -1,128 +1,217 @@
 import React, { useEffect } from "react";
-import { useAccount, useEnsName } from "wagmi";
 import useIsMounted from "../../hooks/useIsMounted";
-import { truncateEthAddress } from "./../../utils";
 
 import { usePayContext } from "../DaimoPay";
 import { TextContainer } from "./styles";
 
 import { AnimatePresence, Variants } from "framer-motion";
-import { Chain } from "viem";
-import { useChainIsSupported } from "../../hooks/useChainIsSupported";
+import { Address, Hex } from "viem";
 import { ResetContainer } from "../../styles";
-import { CustomTheme, Mode, Theme } from "../../types";
+import { CustomTheme, Mode, PaymentOption, Theme } from "../../types";
 import ThemedButton, { ThemeContainer } from "../Common/ThemedButton";
 
-const contentVariants: Variants = {
-  initial: {
-    zIndex: 2,
-    opacity: 0,
-    x: "-100%",
-  },
-  animate: {
-    opacity: 1,
-    x: 0.1,
-    transition: {
-      duration: 0.4,
-      ease: [0.25, 1, 0.5, 1],
-    },
-  },
-  exit: {
-    zIndex: 1,
-    opacity: 0,
-    x: "-100%",
-    pointerEvents: "none",
-    position: "absolute",
-    transition: {
-      duration: 0.4,
-      ease: [0.25, 1, 0.5, 1],
-    },
-  },
-};
+type PayButtonCommonProps =
+  | {
+      /**
+       * Your public app ID. Specify either (payId) or (appId + parameters).
+       */
+      appId: string;
+      /**
+       * Optional nonce. If set, generates a deterministic payID. See docs.
+       */
+      nonce?: bigint;
+      /**
+       * Destination chain ID.
+       */
+      toChain: number;
+      /**
+       * The destination token to send, completing payment. Must be an ERC-20
+       * token or the zero address, indicating the native token / ETH.
+       */
+      toToken: Address;
+      /**
+       * The amount of destination token to send (transfer or approve).
+       */
+      toAmount: bigint;
+      /**
+       * The destination address to transfer to, or contract to call.
+       */
+      toAddress: Address;
+      /**
+       * Optional calldata to call an arbitrary function on `toAddress`.
+       */
+      toCallData?: Hex;
```

Comment:
should all of these be folded into a `recipient`?

```
recipient: {
    address: `0x..`,
    tokenAddress: `0x..`,
    amount: 123n,
    calldata: `0x..`,
}
```

Makes it easier in future if we have something like "Allowed recipients" to add a `recipient: {id: "1234.."}` option

Reply:
one other place where IDs would be useful is bank accounts.

some ideas

### Send $20.00 to a bank account
```
recipient={{
  id: "06b0e3e3-91a6-4b09-bffe-46eab58ffa18",
  currency: "USD",
  quantity: "20.00"
}}
```

vs

```
toAccount="06b0e3e3-91a6-4b09-bffe-46eab58ffa18"
toCurrency="USD"
toQuantity="20.00"
```

Reply:
updated. using `toUnits` to match Viem convention